### ISSUE TITLE
Result.Bind cannot return a different Error type.

### ISSUE THREAD

**Issue by th3oth3rjak3** [th3oth3rjak3/CSharp.Made.Functional#24]
Because Bind does not know how to convert one error type to another, the current implementation does not allow binding to a new error type. Example:

```
public static Result<string, OtherCustomError> SomeBindingOperation(int number) =>
    Result.Failure<string, OtherCustomError>(new OtherCustomError("How do we bind this?"));

Result.Error<int, CustomError>(new CustomError("something bad happened"))
    .Bind(SomeBindingOperation);

```
The binding operation will fail because it can't figure out how to convert OtherCustomError to CustomError.

More research required.

**Comment by th3oth3rjak3** [comment]
I believe this can be solved with a MapError function. Rust uses something like this for mapping error types.

**Comment by th3oth3rjak3** [comment]
Giving this further thought, I think that a BindError method may be useful in cases where the MapError would produce a Result, similar to how Bind is used for when a regular mapping function returns a result.

**Comment by th3oth3rjak3** [comment]
After more thought, I can't think of a valid use case for this. Error mapping should always succeed. Can reevaluate in the future if it is ever needed.

**Comment by th3oth3rjak3** [comment]
Fixed with v 2.1.0
